src/braket/experimental/autoqasm/api.py

```diff
@@ -114,40 +127,50 @@ def gate_calibration(*args, implements: Callable, **kwargs) -> Callable[[], Gate
         `with_calibrations` method of the main program.
     """
     return _function_wrapper(
-        *args,
+        None,
         converter_callback=_convert_calibration,
         converter_args={"gate_function": implements, **kwargs},
     )
 
 
 def _function_wrapper(
-    *args: tuple[Any],
+    func: Optional[Callable],
+    *,
     converter_callback: Callable,
     converter_args: Optional[dict[str, Any]] = None,
-) -> Callable[[Any], aq_program.Program]:
+) -> Callable[..., Optional[Union[aq_program.Program, GateCalibration]]]:
     """Wrapping and conversion logic around the user function `f`.
 
     Args:
+        func (Optional[Callable]): Decorated function. May be `None` in the case where decorator
+            is used with parentheses.
         converter_callback (Callable): The function converter, e.g., _convert_main.
         converter_args (Optional[dict[str, Any]]): Extra arguments for the function converter.
 
     Returns:
-        Callable[[Any], Program]: A callable which returns the converted
-        quantum program when called.
+        Callable[..., Optional[Union[Program, GateCalibration]]]: A callable which
+        returns the converted construct, if any, when called.
     """
-    if not args:
-        # This the case where a decorator is called with only keyword args, for example:
-        #     @aq.main(num_qubits=4)
+    if not (func and callable(func)):
+        # This the case where a decorator is called either without a positional argument,
+        # or with a non-callable positional argument, which is as close of an approximation
+        # we can get to the case where a decorator is called with parentheses.
+        #
+        # There is still a false negative case, where we have something like:
+        #     @aq.main(callable_pos_arg)
         #     def my_function():
+        #
+        # but this is known limitation in python (consider the valid non-decorator usage
+        # `aq.main(my_function)` for an example of why this ambiguity exists).
```

Comment:
brainstorming here (not something we need to do in this PR) - could we ever detect & warn the user if they somehow make this mistake? e.g., the decorated object that is returned could raise some error or warning if it is called again with a function as a first positional argument.

Reply:
I'm not sure exactly which mistake you're referring to. The decorated function returned will (shortly) be an AutoQASM Program, so error handling for calls will be established in that class